R/qenv-join.R

```diff
@@ -23,30 +28,54 @@
 #' qq <- join(q1, q2)
 #' get_code(qq)
 #' @export
-setGeneric("join", function(x, y) {
+setGeneric("join", function(x, y, force, verbose) {
   standardGeneric("join")
 })
 
 #' @rdname join
 #' @export
-setMethod("join", signature = c("qenv", "qenv"), function(x, y) {
+setMethod("join", signature = c("qenv", "qenv"), function(x, y, force, verbose) {
```

Comment:
could force and verbose have default args and then just a simple checkmate::assert_flag() - or does S4 not allow that?

Reply:
+1 for default values,.
Furthermore I suggest `force = TRUE, value = force`.

S4 _does_ allow this. If I'm not mistaken, if assertions are put in a method with `signature = "ANY"` that does not call `callNextMethod()`, they will be applied to all methods with more specific `signature`s.